Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -3499,13 +3499,12 @@ void WebPageProxy::updateFontAttributesAfterEditorStateChange()
 {
     internals().cachedFontAttributesAtSelectionStart.reset();
 
-    if (!internals().editorState.hasPostLayoutData())
+    if (!m_uiClient->needsFontAttributes())
         return;
 
-    if (auto fontAttributes = internals().editorState.postLayoutData->fontAttributes) {
-        m_uiClient->didChangeFontAttributes(*fontAttributes);
-        internals().cachedFontAttributesAtSelectionStart = WTFMove(fontAttributes);
-    }
+    requestFontAttributesAtSelectionStart([this, protectedThis = Ref { *this }] (auto& attributes) {
```

Comment:
nit: style guide suggests no space between capture list and arg list
```suggestion
    requestFontAttributesAtSelectionStart([this, protectedThis = Ref { *this }](auto& attributes) {
```